includes/class-wc-skroutz-analytics-integration.php

```diff
@@ -273,4 +296,160 @@ private function register_widgets() {
 		add_action( 'widgets_init', array( $this, 'register_skroutz_product_reviews_inline_widget') );
 		add_action( 'widgets_init', array( $this, 'register_skroutz_product_reviews_extended_widget') );
 	}
+
+	/**
+	 * Fetches IDs and names of WooCommerce attribute taxonomies.
+	 * @return array [attribute_id => attribute_name] mapping
+	 *
+	 * @since  1.5.0
+	 * @access private
+	 */
+	private function attribute_taxonomies() {
+		return wp_list_pluck( wc_get_attribute_taxonomies(), 'attribute_name', 'attribute_id' );
+	}
+
+	/**
+	 * Generates radio button HTML.
+	 * The `name` attribute of the control should be passed in $data as `name`.
+	 * The currently selected value of the corresponding radio group should be passed in $data as `current_value`.
+	 * @param string $key Field key
+	 * @param array  $data Field data containing the `name` and `current_value` keys
+	 * @return string
+	 *
+	 * @since  1.5.0
+	 * @access private
+	 */
+	private function generate_radio_html( $key, $data ) {
+		$field_key = $this->get_field_key( $key );
+		$defaults  = array(
+			'title'             => '',
+			'label'             => '',
+			'name'              => '',
+			'value'             => '',
+			'current_value'     => '',
+			'disabled'          => false,
+			'class'             => '',
+			'css'               => '',
+			'desc_tip'          => false,
+			'description'       => '',
+			'custom_attributes' => array()
+		);
+
+		$data = wp_parse_args( $data, $defaults );
+
+		if ( ! $data['label'] ) {
+			$data['label'] = $data['title'];
+		}
+
+		ob_start();
+		?>
+		<tr valign="top">
+			<th scope="row" class="titledesc">
+				<label for="<?php echo esc_attr( $field_key ); ?>">
+					<?php echo wp_kses_post( $data['title'] ); ?>
+					<?php echo $this->get_tooltip_html( $data ); // WPCS: XSS ok. ?>
+				</label>
+			</th>
+			<td class="forminp">
+				<fieldset>
+					<legend class="screen-reader-text"><span><?php echo wp_kses_post( $data['title'] ); ?></span></legend>
+					<label for="<?php echo esc_attr( $field_key ); ?>">
+						<input
+							id="<?php echo esc_attr( $field_key ); ?>"
+							name="<?php echo esc_attr( $data['name'] ); ?>"
+							value="<?php echo esc_attr( $data['value'] ); ?>"
+							type="radio"
+							class="<?php echo esc_attr( $data['class'] ); ?>"
+							style="<?php echo esc_attr( $data['css'] ); ?>"
+							<?php checked( $data['current_value'], $data['value'] ); ?>
+							<?php disabled( $data['disabled'], true ); ?>
+							<?php echo $this->get_custom_attribute_html( $data ); // WPCS: XSS ok. ?>
+						/>
+						<?php echo wp_kses_post( $data['label'] ); ?>
+					</label>
+					<br/>
+					<?php echo $this->get_description_html( $data ); // WPCS: XSS ok. ?>
+				</fieldset>
+			</td>
+		</tr>
+		<?php
+
+		return ob_get_clean();
```

Comment:
Could we use Heredoc or Nowdoc strings here instead of output buffering?

Reply:
Within a Heredoc you cannot call a function or use control structures easily